docs/porting/target/i2c.md

```diff
@@ -0,0 +1,114 @@
+## Inter-Integrated Circuit (I2C)
+
+I2C is a serial protocol for two-wire interface to connect low-speed devices
+in embedded systems. The I2C API allows control and configuration of this
+interface.
+
+The interface is made up of two lines for all communication:
+  - SCL: Serial Clock
+  - SDA: Serial Data
+
+### Assumptions
+
+#### Defined Behaviours
+
+- `i2c_init`:
+  - Initialises the peripheral pins specified in the input parameters.
+  - Initialises the peripheral in master mode if `is_slave` is false.
+  - Initialises the peripheral in slave mode if `is_slave` is true and `supports_slave_mode` is true.
```

Comment:
Can we add
"- Initializes all object fields to 0"?
The fact that this isn't done I noticed can cause significant problems when calling i2c_init directly without using the higher level I2C driver function (mbed-os issue 9890)